Updated for Elm 0.17

Kyarigwo opened the pull request:
I have changed the header in generated modules to read 'exposing(..)' rather then 'where', in line with Elm version 0.17.

However, should the be able to be switched for backward compatibility?

agrafix commented:
Thanks! Yes, I think for now it is important to stay backwards compatible, so adding an option to select the target elm version would be great.

Kyarigwo commented:
I've created an issue for this update with the below comment.

I have updated my pull request for updating to Elm version 0.17.  The changes that I have made are:

Add a new module, Elm.Versions exporting a type 
data ElmVersion
  = Elm0p16
  | Elm0p17

Modify Elm.Module, adding a function 
MakeElmModuleWithVersion which takes an ElmVersion as it's first parameter, hence switching on version, and modified the existing makeElmModule to be makeElmModuleWithVersion Elm0p16, so that existing code should run without any changes.

Finally, I have added two tests to ModuleSpec, one to test makeElmModuleWithVersion for Elm0p16 and one for Elm0p17.

Please suggest any changes.

agrafix commented:
Thanks! Looks good to me.
